### ISSUE TITLE
Game executable deletes itself upon failures on Windows

### ISSUE THREAD

**Issue by Qix-** [OpenRCT2/OpenRCT2#7768]
**OS:** Windows 10
**Version:** 1803
**Commit/Build:** 0aff800

On Windows, running the game crashes in various scenarios. When this happens, the original with-icon .exe is replaced with an MS-DOS (non-icon, console mode) executable, and then upon a _second_ crash the executable is removed altogether.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Download and install/unpack
2. Run the game; crash somehow (e.g. try to use the portable exe in latest dev (3f8c3c3) or simply try to use multiplayer in latest release) - game is replaced with strange MS-DOS version.
3. Repeat step 2 exactly once more, game executable (_not_ accompanying installation files) is removed entirely.

**Dump file**

None.

**Screenshots / Video:**

None.

**Save game:**

None.

**Comment by janisozaur** [comment]
We don't ship MS-DOS version.

There is no code in OpenRCT2 to remove itself, but we've heard from multiple users on Windows who had some weird antivirus ~~mal~~software installed about having such issues. Do you have any such thing installed and running?

> running the game crashes in various scenarios. 

This sounds like separate issues and please file each one accordingly.

**Comment by IntelOrca** [comment]
@janisozaur What he said was correct, the .com file shows up as an MS-DOS program / icon even though it isn't.

**Comment by Qix-** [comment]
Correct, sorry should have been more specific: it's not an MS-DOS version of the exe as that would be kind of funny. The icon resource just seems to get stripped. Also, the filetype in Explorer shows MS-DOS executable instead of application or whatever.

Also, I should have mentioned it a bit more in the OP - the replaced version (the second run, with the faux-MS-DOS version) allocates a console whereas the original executable does not. It's like the subsystem flags get changed in the executable itself, which is a weird thing for AV to do (I would think).

Seeing as how that didn't strike me as AV behavior and we got absolutely no AV notifications (I don't know whether or not that particular machine had AV - I'll check) that's why it didn't immediately seem like an AV issue.

I'll check and report back. Thanks for the responses.

**Comment by IntelOrca** [comment]
@Qix- We ship two binaries, openrct2.exe and openrct2.com. The only difference between them is the subsystem, this is so that when `openrct2` is invoked on the console, stdin / stdout works properly without forcing all players to have a console window showing when running the game normally.
